invenio/base/factory.py

```diff
@@ -155,22 +156,44 @@ def configure_warnings():
         warnings.simplefilter("default", DeprecationWarning)
 
 
-def create_app(instance_path=None, **kwargs_config):
+def create_app(instance_path=None, static_folder=None, **kwargs_config):
     """Prepare Invenio application based on Flask.
 
     Invenio consists of a new Flask application with legacy support for
     the old WSGI legacy application and the old Python legacy
     scripts (URLs to ``*.py`` files).
+
+    For configuration variables detected from environment variables, a prefix
+    will be used which is the uppercase version of the app name, excluding
+    any non-alphabetic ('[^A-Z]') characters.
+
+    If `instance_path` is `None`, the `<PREFIX>_INSTANCE_PATH` environment
+    variable will be used. If that one does not exist, a path inside
+    `sys.prefix` will be used.
+
+    If `static_folder` is `None`, the `<PREFIX>_STATIC_FOLDER` environment
+    variable will be used. If that one does not exist, a path inside the
+    detected `instance_path` will be used.
     """
     configure_warnings()
 
     # Flask application name
     app_name = '.'.join(__name__.split('.')[0:2])
 
-    # Force instance folder to always be located in under system prefix
-    instance_path = instance_path or os.path.join(
-        sys.prefix, 'var', app_name + '-instance'
-    )
+    # Prefix for env variables
+    env_prefix = re.sub('[^A-Z]', '', app_name.upper())
+
+    # Detect instance path
+    instance_path = instance_path or \
+        os.getenv(env_prefix + '_INSTANCE_PATH') or \
```

Comment:
We should discuss potential security issues when using OS environment variables.

--
cc @lnielsen @tiborsimko @jalavik